container/go/pkg/oci/reader_test.go

```diff
@@ -0,0 +1,98 @@
+package oci
+
+import (
+	"testing"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/google/go-containerregistry/pkg/v1/validate"
+)
+
+// This test index is pulled by puller.go from gcr.io/distroless/base@sha256:edc3643ddf96d75032a55e240900b68b335186f1e5fea0a95af3b4cc96020b77
+var (
```

Comment:
Suggest using table driven tests here. See https://github.com/golang/go/wiki/TableDrivenTests#example-of-a-table-driven-test. Create an anonymous struct that stores the variables you are storing here. This way we can add more test cases easily in the future that test more types of images.